marketplaceItems/drawApp/fingerDraw_app.js

```diff
@@ -0,0 +1,596 @@
+//
+//  draw_app.js
+//
+//  Created by Rebecca Stankus on 01/31/19
+//  Copyright 2019 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+/* global  Audio, Camera, Controller, Entities, HMD, Messages, MyAvatar, Quat, Script, Settings, SoundCache, 
+Tablet, Vec3, Window */
+
+(function() {
+    var REPEAT_DISTANCE_CHECK_MS = 60;
+    var MINIMUM_MOVEMENT_TO_DRAW_M = 0.0005;
+    var DEFAULT_NORMAL = { x: 0, y: 0, z: 1 };
+    var DECAY_TIME_S = 60;
+    var MAX_LINE_POINTS = 100;
+    var DRAW_SOUND = SoundCache.getSound(Script.resolvePath('assets/sounds/markerDraw.mp3'));
+    var DRAW_SOUND_VOLUME = 0.01;
+    var distanceCheckInterval = null;
+    var polyLine = null;
+    var lineStartPosition;
+    var previousLinePoint;
+    var linePoints = [{x: 0, y: 0, z: 0 }];
+    var lineNormals = [DEFAULT_NORMAL, DEFAULT_NORMAL];
+    var lineStrokeWidths = [];
+    var paintSphere;
+    var parentJointIndex;
+    var activeTriggerPress = false;
+    var activeGripPress = false;
+    var paintSphereDimensions;
+
+    // *************************************
+    // START UTILITY FUNCTIONS
+    // *************************************
+
+    /* PLAY A SOUND: Plays the specified sound at the position of the user's Avatar using the volume and playback 
+    mode requested. */
+    var injector;
+    function playSound(sound, volume, position, localOnly, loop){
+        if (sound.downloaded) {
+            if (injector) {
+                injector.stop();
+                injector = null;
+            }
+            injector = Audio.playSound(sound, {
+                position: position,
+                volume: volume,
+                localOnly: localOnly,
+                loop: loop
+            });
+        }
+    }
+
+    // *************************************
+    // END UTILITY FUNCTIONS
+    // *************************************
+
+    /* CREATE A PAINTBALL: Checks that paint sphere does not already exist, then calculate position of avatar's hand and 
+    create a paint sphere there */
+    function createPaintSphere() {
+        if (paintSphere) {
+            return;
+        }
+        var fingertipJointName = dominantHandJoint + "Index4";
+        parentJointIndex = MyAvatar.getJointIndex(fingertipJointName);
+        if (parentJointIndex === -1) {
+            fingertipJointName = dominantHandJoint + "Index3";
+            parentJointIndex = MyAvatar.getJointIndex(fingertipJointName);
+        }
```

Comment:
if, at this point, the avatar doesn't have either of these two joints - what do you think about falling back to the hand joint and printing a log to the console?

Reply:
I thought we couldn't parent to hand joints, but I will test and see if it works. I think it might now...especially since we are disabling the controller.

Reply:
Seems to work fine!